clientd3d/d3drender_objects.h

```diff
@@ -13,6 +13,150 @@
 #ifndef _D3DRENDEROBJECTS_H
 #define _D3DRENDEROBJECTS_H
 
-void D3DRenderObjects(room_type* room, Draw3DParams* params, room_contents_node* pRNode);
+struct PlayerViewParams
+{
+    // Player information
+    player_info* player;
+
+    // Screen dimensions
+    int screenWidth;
+    int screenHeight;
+
+    // Viewport dimensions
+    int viewportWidth;
+    int viewportHeight;
+
+    // D3D viewport rectangle
+    RECT d3dRect;
+
+    PlayerViewParams(
+        player_info* playerParam,
+        int screenWidthParam,
+        int screenHeightParam,
+        int viewportWidthParam,
+        int viewportHeightParam,
+        RECT d3dRectParam)
+        : player(playerParam),
+          screenWidth(screenWidthParam),
+          screenHeight(screenHeightParam),
+          viewportWidth(viewportWidthParam),
+          viewportHeight(viewportHeightParam),
+          d3dRect(d3dRectParam)
+    {}
+};
+
+struct FontTextureParams
+{
+    font_3d* font;
+    Color* basePalette;
+    int smallTextureSize;
+
+    FontTextureParams(
+        font_3d* fontParam,
+        Color* basePaletteParam,
+        int smallTextureSizeParam)
+        : font(fontParam),
+          basePalette(basePaletteParam),
+          smallTextureSize(smallTextureSizeParam)
+    {}
+};
+
+struct LightAndTextureParams
+{
+    // Light cache details
+    d_light_cache* lightCache;
+    d_light_cache* lightCacheDynamic;
+
+    // Texture and sector settings
+    int smallTextureSize;
+    int* sectorDepths;
+
+    LightAndTextureParams(
+        d_light_cache* lightCacheParam,
+        d_light_cache* lightCacheDynamicParam,
+        int smallTextureSizeParam,
+        int* sectorDepthsParam)
+        : lightCache(lightCacheParam),
+          lightCacheDynamic(lightCacheDynamicParam),
+          smallTextureSize(smallTextureSizeParam),
+          sectorDepths(sectorDepthsParam)
+    {}
+};
+
+struct GameObjectDataParams
+{
+    // Information about draw items and object ranges
+    long numItems;
+    int& numVisibleObjects;
+    int& numObjects;
+    DrawItem* drawData;
+    ObjectRange* visibleObjects;    // Where objects are on screen
+
+    // Textures for the back buffer
+    PDIRECT3DTEXTURE9 backBufferTexFull;
+    LPDIRECT3DTEXTURE9 (&backBufferTex)[16];
+
+    GameObjectDataParams(
+        long numItemsParam,
+        int& numVisibleObjectsParam,
```

Comment:
I'm not too excited about these as references, because it hides the fact that these are pointers rather than just ints.    If there's no way to avoid having them here, I guess convert to pointers?